src/supertux/player_status.cpp

```diff
@@ -301,17 +377,55 @@ PlayerStatus::remove_player(int player_id)
   for (int i = player_id; i < m_num_players; i++)
   {
     bonus[i] = bonus[i + 1];
-    max_fire_bullets[i] = max_fire_bullets[i + 1];
-    max_ice_bullets[i] = max_ice_bullets[i + 1];
-    max_air_time[i] = max_air_time[i + 1];
-    max_earth_time[i] = max_earth_time[i + 1];
+    m_item_pockets[i] = m_item_pockets[i + 1];
   }
 
   bonus.resize(m_num_players, NO_BONUS);
-  max_fire_bullets.resize(m_num_players, 0);
-  max_ice_bullets.resize(m_num_players, 0);
-  max_air_time.resize(m_num_players, 0);
-  max_earth_time.resize(m_num_players, 0);
+  m_item_pockets.resize(m_num_players, NO_BONUS);
+}
+
+PlayerStatus::PocketPowerUp::PocketPowerUp(BonusType bonustype, Vector pos):
+  PowerUp(pos, PowerUp::get_type_from_bonustype(bonustype)),
+  m_cooldown_timer(),
+  m_blink_timer(),
+  m_visible(true)
```

Comment:
Simply suggesting another newline here for some separation from the `PlayerStatus` class.

```suggestion


PlayerStatus::PocketPowerUp::PocketPowerUp(BonusType bonustype, Vector pos):
  PowerUp(pos, PowerUp::get_type_from_bonustype(bonustype)),
  m_cooldown_timer(),
  m_blink_timer(),
  m_visible(true)
```